app/src/main/java/com/example/criminalintent/presentation/crime_detail/CrimeFragment.kt

```diff
@@ -50,14 +51,51 @@ class CrimeFragment : Fragment(), FragmentResultListener {
         super.onViewCreated(view, savedInstanceState)
         childFragmentManager.setFragmentResultListener(DIALOG_DATE, viewLifecycleOwner, this)
         observeData()
+        setOnBackPressed()
+    }
+
+    private fun loadCrimeFromArguments() {
+        val crimeId: Int = arguments?.getSerializable(ARG_CRIME_ID) as Int
+        crimeDetailViewModel.loadCrime(crimeId)
+    }
+
+    private fun setOnBackPressed() {
+        activity?.onBackPressedDispatcher?.addCallback(
+            viewLifecycleOwner,
+            object : OnBackPressedCallback(true) {
+                override fun handleOnBackPressed() {
+                    showAlertDialog()
+                }
+            }
+        )
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        super.onCreateOptionsMenu(menu, inflater)
+        inflater.inflate(R.menu.view_crimedetail, menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            R.id.add_crime -> {
+                addOrUpdateCrime()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun addOrUpdateCrime() {
```

Comment:
This function is named `addOrUpdateCrime` it should do just that. Move the pop back stack out of it

Reply:
https://www.oodesign.com/single-responsibility-principle.html